docs/large_scale.md

```diff
@@ -0,0 +1,136 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.9.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+:::{currentmodule} tsinfer
+:::
+
+(sec_large_scale)=
+
+# Large Scale Inference
+
+tsinfer scales well and has been successfully used with datasets up to half a
+million samples. Here we detail considerations and tips for each step of the
+inference process to help you scale up your analysis. A snakemake pipeline
+which implements this parallelisation scheme is available at https://github.com/benjeffery/tsinfer-snakemake.
+
+(sec_large_scale_ancestor_generation)=
+
+## Data preparation
+
+For large scale inference the data must be in [VCF Zarr](https://github.com/sgkit-dev/vcf-zarr-spec)
+format, read by the {class}`VariantData` class. [bio2zarr](https://github.com/sgkit-dev/bio2zarr)
+is recommended for conversion from VCF. [sgkit](https://github.com/sgkit-dev/sgkit) can then
+be used to perform initial filtering.
+
+
+## Ancestor generation
+
+Ancestor generation is generally the fastest step in inference and is not yet
+parallelised out-of-core in tsinfer. However it scales well on machines with
+many cores and hyperthreading via the `num_threads` argument to
+{meth}`generate_ancestors`. The limiting factor is often that the
+entire genotype array for the contig being inferred needs to fit in RAM.
+This is the high-water mark for memory usage in tsinfer.
+Note the `genotype_encoding` argument, setting this to
+{class}`tsinfer.GenotypeEncoding.ONE_BIT` reduces the memory footprint of
+the genotype array by a factor of 8, for a surprisingly small increase in
+runtime. With this encoding, the RAM needed is roughly 
+`num_sites * num_samples * ploidy / 8 bytes.`
+
+## Ancestor matching
+
+Ancestor matching is one of the more time consuming steps of inference. It
+proceeds in groups, progressively growing the tree sequence with younger
+ancestors. At each stage the parallelism is limited to the number of ancestors
+whose possible inheritors are already matched, as all possible inheritors
+of a sample must be matched in an earlier group. For a typical human data set
+the number of samples per group varies from single digits up to approximately
+the number of samples.
+The plot below shows the number of ancestors matched in each group for a typical
+human data set:
+
+```{figure} _static/ancestor_grouping.png
```

Comment:
May be worth indicating that the group number is ordered by time, so that group 0 represents the oldest ancestors?

Reply:
Fixed in 6c02df1